devito/types/dense.py

```diff
@@ -1311,3 +1312,198 @@ def parent(self):
         return self._parent
 
     _pickle_kwargs = Function._pickle_kwargs + ['parent']
+
+
+class SeparableFunction(Differentiable):
+    """
+    A function separable in its dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
+
+    Takes the same parameters as Function and
+
+    Parameters
+    ----------
+        sep: 'sum' or 'prod'
+            type of separation as a product or sum of function
+        separated: tuple or Dimension
+            Dimension in which the function is separable
+        rank: Int
+            Size of the subfunction factors
+
+    Examples
+    --------
+
+    Creation
+
+    >>> from devito import Grid, TimeFunction
+    >>> grid = Grid(shape=(4, 4, 5))
+    >>> x, y, z = grid.dimensions
+    >>> a = SeparableFunction(name='a', grid=grid)
```

Comment:
No need to assign it to ``a``. Same for the lines below. Same for the SeparableTimeFunction docstring